File: kumi/sdk/java/src/main/java/io/kumi/RegisterOptions.java
package io.kumi;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

/**
 * Builder-style configuration for registering a tool with KumiAgent.
 */
public class RegisterOptions {
    private String name;
    private String description;
    private List<ToolParameter> parameters = new ArrayList<>();
    private Integer timeout;
    private boolean requireConfirmation;
    private boolean alwaysInclude;
    private boolean allowProactive;
    private boolean proactiveContext;
    private Map<String, Object> proactiveContextArgs;
    private String proactiveContextDescription;
    private ToolHandler handler;

    public RegisterOptions name(String name) { this.name = name; return this; }
    public RegisterOptions description(String desc) { this.description = desc; return this; }
    public RegisterOptions parameters(ToolParameter... params) { this.parameters = Arrays.asList(params); return this; }
    public RegisterOptions parameters(List<ToolParameter> params) { this.parameters = params; return this; }
    public RegisterOptions timeout(int seconds) { this.timeout = seconds; return this; }
    public RegisterOptions requireConfirmation(boolean v) { this.requireConfirmation = v; return this; }
    public RegisterOptions alwaysInclude(boolean v) { this.alwaysInclude = v; return this; }
    public RegisterOptions allowProactive(boolean v) { this.allowProactive = v; return this; }
    public RegisterOptions proactiveContext(boolean v) { this.proactiveContext = v; return this; }
    public RegisterOptions proactiveContextArgs(Map<String, Object> v) { this.proactiveContextArgs = v; return this; }
    public RegisterOptions proactiveContextDescription(String v) { this.proactiveContextDescription = v; return this; }
    public RegisterOptions handler(ToolHandler h) { this.handler = h; return this; }

    public String getName() { return name; }
    public String getDescription() { return description; }
    public List<ToolParameter> getParameters() { return parameters; }
    public Integer getTimeout() { return timeout; }
    public boolean isRequireConfirmation() { return requireConfirmation; }
    public boolean isAlwaysInclude() { return alwaysInclude; }
    public boolean isAllowProactive() { return allowProactive; }
    public boolean isProactiveContext() { return proactiveContext; }
    public Map<String, Object> getProactiveContextArgs() { return proactiveContextArgs; }
    public String getProactiveContextDescription() { return proactiveContextDescription; }
    public ToolHandler getHandler() { return handler; }
}

File: kumi/sdk/csharp/RegisterOptions.cs
using System.Collections.Generic;

namespace Kumi;

/// <summary>
/// Builder-style configuration for registering a tool with <see cref="KumiAgent"/>.
/// </summary>
public sealed class RegisterOptions
{
    public string? Name { get; private set; }
    public string? Description { get; private set; }
    public List<ToolParameter> Parameters { get; private set; } = new();
    public int? Timeout { get; private set; }
    public bool RequireConfirmation { get; private set; }
    public bool AlwaysInclude { get; private set; }
    public bool AllowProactive { get; private set; }
    public bool ProactiveContext { get; private set; }
    public Dictionary<string, object>? ProactiveContextArgs { get; private set; }
    public string? ProactiveContextDescription { get; private set; }
    public ToolHandler? Handler { get; private set; }

    public RegisterOptions SetName(string name) { Name = name; return this; }
    public RegisterOptions SetDescription(string desc) { Description = desc; return this; }
    public RegisterOptions SetParameters(params ToolParameter[] pars) { Parameters = new List<ToolParameter>(pars); return this; }
    public RegisterOptions SetTimeout(int seconds) { Timeout = seconds; return this; }
    public RegisterOptions SetRequireConfirmation(bool v) { RequireConfirmation = v; return this; }
    public RegisterOptions SetAlwaysInclude(bool v) { AlwaysInclude = v; return this; }
    public RegisterOptions SetAllowProactive(bool v) { AllowProactive = v; return this; }
    public RegisterOptions SetProactiveContext(bool v) { ProactiveContext = v; return this; }
    public RegisterOptions SetProactiveContextArgs(Dictionary<string, object> args) { ProactiveContextArgs = args; return this; }
    public RegisterOptions SetProactiveContextDescription(string description) { ProactiveContextDescription = description; return this; }
    public RegisterOptions SetHandler(ToolHandler h) { Handler = h; return this; }
}
